pkg/kubectl/cmd/auth/cani.go

```diff
@@ -167,7 +185,22 @@ func (o *CanIOptions) Validate() error {
 	return nil
 }
 
-func (o *CanIOptions) RunAccessCheck() (bool, error) {
+func (o *CanIOptions) RunAccessCheck() error {
+	if o.List {
```

Comment:
I'd imagine a separate function called from here for this functionality, it will read better.